drivers/ethernet/eth_xlnx_gem.c

```diff
@@ -0,0 +1,1724 @@
+		dev_conf->hw_rx_buffer_size == HWRX_BUFFER_SIZE_1KB),
+		"%s hardware RX buffer size configuration is invalid",
+		dev->name);
+
+	/* HW RX buffer offset */
+	__ASSERT(dev_conf->hw_rx_buffer_offset <= 3,
+		"%s hardware RX buffer offset %u is invalid, must be "
+		"in range 0 to 3", dev->name, dev_conf->hw_rx_buffer_offset);
+
+	/*
+	 * RX & TX buffer sizes
+	 * RX Buffer size must be a multiple of 64, as the size of the
+	 * corresponding DMA receive buffer in AHB system memory is
+	 * expressed as n * 64 bytes in the DMA configuration register.
+	 */
+	__ASSERT(dev_conf->rx_buffer_size % 64 == 0,
+		"%s RX buffer size %u is not a multiple of 64 bytes",
+		dev->name, dev_conf->rx_buffer_size);
+	__ASSERT((dev_conf->rx_buffer_size != 0 &&
+		dev_conf->rx_buffer_size <= 16320),
+		"%s RX buffer size %u is invalid, should be >64, "
+		"must be 16320 bytes maximum.", dev->name,
+		dev_conf->rx_buffer_size);
+	__ASSERT((dev_conf->tx_buffer_size != 0 &&
+		dev_conf->tx_buffer_size <= 16380),
+		"%s TX buffer size %u is invalid, should be >64, "
+		"must be 16380 bytes maximum.", dev->name,
+		dev_conf->tx_buffer_size);
+
+	/*
+	 * Initialization procedure as described in the Zynq-7000 TRM,
+	 * chapter 16.3.x.
+	 */
+	eth_xlnx_gem_reset_hw(dev);		/* Chapter 16.3.1 */
+	eth_xlnx_gem_set_initial_nwcfg(dev);	/* Chapter 16.3.2 */
+	eth_xlnx_gem_set_mac_address(dev);	/* Chapter 16.3.2 */
+	eth_xlnx_gem_set_initial_dmacr(dev);	/* Chapter 16.3.2 */
+
+	/* Enable MDIO -> set gem.net_ctrl[mgmt_port_en] */
+	if (dev_conf->init_phy == 1) {
+		reg_val = sys_read32(
+			dev_conf->base_addr + ETH_XLNX_GEM_NWCTRL_OFFSET);
+		reg_val |= ETH_XLNX_GEM_NWCTRL_MDEN_BIT;
+		sys_write32(reg_val,
+			dev_conf->base_addr + ETH_XLNX_GEM_NWCTRL_OFFSET);
+	}
+
+	eth_xlnx_gem_configure_clocks(dev);	/* Chapter 16.3.3 */
+	if (dev_conf->init_phy == 1) {
+		eth_xlnx_gem_init_phy(dev);	/* Chapter 16.3.4 */
+	}
+	eth_xlnx_gem_configure_buffers(dev);	/* Chapter 16.3.5 */
+
+	return 0;
+}
+
+/**
+ * @brief GEM associated interface initialization function
+ * Initializes the interface associated with a GEM device.
+ *
+ * @param iface Pointer to the associated interface data struct
+ */
+static void eth_xlnx_gem_iface_init(struct net_if *iface)
+{
+	const struct device *dev = net_if_get_device(iface);
+	const struct eth_xlnx_gem_dev_cfg *dev_conf = DEV_CFG(dev);
+	struct eth_xlnx_gem_dev_data *dev_data = DEV_DATA(dev);
+
+	/* Set the initial contents of the current instance's run-time data */
+	dev_data->iface = iface;
+	net_if_set_link_addr(iface, dev_data->mac_addr, 6, NET_LINK_ETHERNET);
+	ethernet_init(iface);
+	net_if_flag_set(iface, NET_IF_NO_AUTO_START);
+
+	/*
+	 * Initialize the (delayed) work items for RX pending, TX done
+	 * and PHY status polling handlers
+	 */
+	k_work_init(&dev_data->tx_done_work, eth_xlnx_gem_tx_done_work);
+	k_work_init(&dev_data->rx_pend_work, eth_xlnx_gem_rx_pending_work);
+	k_delayed_work_init(&dev_data->phy_poll_delayed_work,
```

Comment:
This work API is deprecated and replaced by `k_work_init_delayable`, this needs to be converted before the driver can be merged.

Reply:
The delayable work API calls have been updated to the current version.